app/Http/Controllers/OneclickMallController.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Transbank\Webpay\Options;
+use Transbank\Webpay\Oneclick\MallInscription;
+use Transbank\Webpay\Oneclick\MallTransaction;
+
+class OneclickMallController extends Controller
+{
+
+    const  TIMEOUT = -96;
+    const REJECTED = -1;
+    const COMMERCE_CODE = "597055555541";
+    const API_KEY = "579B532A7440BB0C9079DED94D31EA1615BACEB56610332264630D42D0A36B1C";
+    private MallInscription $mallInscription;
+    private MallTransaction $mallTransaction;
+
+    public function __construct()
+    {
+        $option = new Options(self::API_KEY, self::COMMERCE_CODE, Options::ENVIRONMENT_INTEGRATION);
+        $this->mallInscription = new MallInscription($option);
+        $this->mallTransaction = new MallTransaction($option);
+    }
+
+    public function startInscription()
+    {
+        try {
+
+            $startTx = [
+                "username" => "User-" . random_int(1, 10000),
+                "email" => "user." . random_int(1, 10000) . "@example.cl",
+                "response_url" => url("/") . "/oneclick-mall/finish"
+            ];
+
+            session(['username' => $startTx["username"]]);
+            $resp = $this->mallInscription->start($startTx["username"], $startTx["email"], $startTx["response_url"]);
+            return view('oneclick-mall.start', ["request" => $startTx, "resp" => $resp]);
+        } catch (\Exception $e) {
+            $error = ["msg" => $e->getMessage(), "code" => $e->getCode()];
+            return view('error-page', ["error" => $error]);
+        }
+    }
+
+    public function finishInscription(Request $request)
+    {
+        try {
+            $view = 'error-page';
+            $data = ["error" => $request];
+            $params = $request->only(['TBK_ORDEN_COMPRA', 'TBK_TOKEN', 'TBK_ID_SESION']);
+            $token = $request["TBK_TOKEN"];
+            $userName = session('username', '');
+
+            if ($request->exists("TBK_ORDEN_COMPRA")) {
+                return view('oneclick-mall.recoverTransaction', ["req" => $params]);
+            }
+
+            $resp = $this->mallInscription->finish($token);
+
+            if ($resp->responseCode == self::REJECTED) {
+                $view = 'oneclick-mall.rejected';
+                $data = ["resp" => $resp, "token" => $token];
+                return view('oneclick-mall.rejected', ["resp" => $resp, "token" => $token]);
+            } else if ($resp->responseCode == self::TIMEOUT) {
```

Comment:
use `elseif` instead of `else if`
```suggestion
            } elseif ($resp->responseCode == self::TIMEOUT) {
```